flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,719 @@
+
+import re
 import logging
+import feedparser
 
 from flexget import plugin
 from flexget.entry import Entry
 from flexget.event import event
-import feedparser
+from flexget.config_schema import one_or_more
+from flexget.utils.tools import parse_timedelta, str_to_boolean, split_title_year
+from flexget.utils.search import normalize_unicode
 
-__author__ = 'deksan'
+__author__ = 'andy (org. by deksan)'
 
 log = logging.getLogger('newznab')
 
+CATEGORIES = {
+    # Games Consoles
+    'game': 1000,
+    'game/nds': 1010,
+    'game/psp': 1020,
+    'game/wii': 1030,
+    'game/xbox': 1040,
+    'game/xbox360': 1050,
+    'game/wiiware': 1060,
+    'game/xbox360dlc': 1070,
+    'game/ps3': 1080,
+    'game/xbox-one': 1090,  # unofficial
+    'game/ps4': 1100,  # unofficial
+    # Movie
+    'movie': 2000,
+    'movie/foreign': 2010,
+    'movie/other': 2020,
+    'movie/sd': 2030,
+    'movie/hd': 2040,
+    'movie/uhd': 2045,  # unofficial
+    'movie/bluray': 2050,
+    'movie/3d': 2060,
+    'movie/dvd': 2070,
+    'movie/web-dl': 2080,  # unofficial
+    # Music
+    'music': 3000,
+    'music/mp3': 3010,
+    'music/video': 3020,
+    'music/audiobook': 3030,
+    'music/lossless': 3040,
+    'music/foreign': 3060,  # unofficial
+    # PC (Games + Software)
+    'pc': 4000,
+    'pc/0day': 4010,
+    'pc/iso': 4020,
+    'pc/mac': 4030,
+    'pc/mobile-other': 4040,
+    'pc/games': 4050,
+    'pc/mobile-ios': 4060,
+    'pc/mobile-android': 4070,
+    # TV
+    'tv': 5000,
+    'tv/web-dl': 5010,  # unofficial
+    'tv/foreign': 5020,
+    'tv/sd': 5030,
+    'tv/hd': 5040,
+    'tv/uhd': 5045,  # unofficial
+    'tv/other': 5050,
+    'tv/sport': 5060,
+    'tv/anime': 5070,
+    'tv/documentary': 5080,  # unofficial
+    # XXX
+    'xxx': 6000,
+    'xxx/dvd': 6010,
+    'xxx/wmv': 6020,
+    'xxx/xvid': 6030,
+    'xxx/x264': 6040,
+    'xxx/uhd': 6045,  # unofficial
+    'xxx/pack': 6050,
+    'xxx/imageset': 6060,
+    'xxx/packs': 6070,  # unofficial
+    'xxx/sd': 6080,  # unofficial
+    'xxx/web-dl': 6090,  # unofficial
+    # Misc
+    'misc': 7000,
+    'misc/misc': 7010,
+    'misc/ebook': 7020,
+    'misc/comics': 7030,
+    # Unknown
+    'not-determined': 7900
+}
+
+# NOTE: all lowercase only
+NAMESPACE_ATTRIBUTE_MAP = {
+    'grabs': int,
+    'size': int,
+    'files': int,
+    'usenetdate': datetime,
+    'password': bool,
+    'guid': basestring,
+    'hydraindexername': basestring,
+    'hydraindexerhost': basestring,
+    'hydraindexerscore': int
+}
+
+PLUGIN_LOOKUP_MAP = {
+    'tvdb': 'thetvdb_lookup',
+    'trakt': 'trakt_lookup',
+    'tvmaze': 'tvmaze_lookup',
+    'imdb': 'imdb_lookup',
+    'tmdb': 'tmdb_lookup'
+}
+
+NAMESPACE_NAME = 'newznab'
+NAMESPACE_URL = 'http://www.newznab.com/DTD/2010/feeds/attributes/'
+NAMESPACE_TAGNAME = 'attr'
+NAMESPACE_PREFIX = 'newznab_'
+ENCLOSURE_TYPE = 'application/x-nzb'
+
+
+# utils
+def list_insert_unique(in_list, idx, element, caseinsensitive=False):
```

Comment:
Check python set

It's not case insensitive though by nature, but it would be a LOT more elegant if you could utilize that instead as a basis.